### ISSUE TITLE
The SAM adapter's image encoder runs with autograd on, and its output is cached

### ISSUE THREAD

**Issue by JArmandoAnaya** [Robomous/VisionSet#613]
The SAM adapter's image encoder — the expensive half of a point suggestion, and the one whose
result is deliberately cached — runs with autograd enabled. Every other forward pass in the
package runs inside `_fp16.forward_guard`, which exists partly to supply `torch.no_grad()`;
this one is called before that guard is entered and is covered by nothing.

## Evidence at `f31653d`

`src/visionset/inference/sam_provider.py:325-337`:

```python
        with self._encoding.for_key(target.asset_id):
            held = self._embeddings.get(target.asset_id)
            if held is not None:
                return held
            image = Image.open(BytesIO(target.content)).convert("RGB")
            size = (image.height, image.width)
            inputs = processor(images=image, return_tensors="pt").to(device)
            self._encodes += 1
            return self._embeddings.put(
                target.asset_id, (model.get_image_embeddings(inputs["pixel_values"]), size)
            )
```

`model.get_image_embeddings(...)` is a real forward through the vision encoder. It is not
wrapped, and nothing above it wraps it either: `_one` at `sam_provider.py:245-259` calls
`self._embedding(...)` on line 246 and only enters the guard on line 254, for the mask decoder:

```python
        processor, model, device, half = self._ready()
        embedding, size = self._embedding(target, processor=processor, model=model, device=device)
        ...
        with _fp16.forward_guard(torch, device_type=device.split(":")[0], half=half):
            outputs = model(...)
```

`forward_guard` (`src/visionset/inference/_fp16.py:76-105`) states the rule the encode path is
missing: *"`no_grad` always: nothing here trains, and gradients on a batch of images are the
difference between fitting in 16 GB and not."*

The sibling adapter is correct — `transformers_provider.py:213-214` has its only forward inside
the guard — so this is one site, not a package-wide posture.

## Why it matters

Two costs, and the second is the one that persists.

1. **During the encode**, the encoder builds an autograd graph it will never use. That is extra
   device memory at the exact moment the largest allocation of the request is live.
2. **After the encode**, the returned tensor carries `grad_fn`, so putting it in
   `self._embeddings` (`BoundedCache`, `DEFAULT_EMBEDDING_CAPACITY`) pins the encoder's
   intermediate activations for as long as that entry survives. The cache is bounded by
   *count*, and the documented budget for an entry is the embedding alone — `KeyedLocks`'
   docstring reasons about "the 16 MB the same asset's embedding occupies while it is cached".
   With a graph attached, an entry is not 16 MB, and the bound is being computed against the
   wrong number.

This is a plain resource defect rather than a wrong answer: predictions are unaffected.

## Reachability

Every point suggestion on a not-yet-encoded asset, which is the first click on every frame. The
curated point-prompt list now includes a checkpoint roughly an order of magnitude larger than
the balanced rung, so the headroom this consumes is headroom that is already contested —
cf. #610, which is about the failure at the other end of the same budget.

## Suggested shape

Route the encode through the same guard the decode uses, rather than adding a second
`torch.no_grad()` at this call site — the package already made the move of putting device and
precision handling in one helper, and a second spelling is what that consolidation exists to
prevent. Note that `forward_guard` also installs the `grid_sample` dtype shim; whether the
encoder needs that on the half-precision path should be answered rather than assumed.

A test can assert this cheaply without a GPU: the existing stub-torch tests in
`tests/inference/test_fp16.py` already prove the guard is entered and restored, so the encode
path can be held to the same assertion.

**Severity: moderate.** No incorrect output; wasted device memory on every cold click, and a
cache whose per-entry cost is larger than the bound was sized for.
